tests: refactor tests

ktaborowski opened the pull request:
[KRKNWK-14886]
Move and refactor tests:

- [x] critical_region
- [ ] crypto
- [x] crypto_keys
- [x] delay
- [x] interrupts
- [ ] ~~mfg~~ _not working_
- [ ] spi
- [ ] storage
- [x] temperature
- [x] time

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: 

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [x] Tests were updated (if applicable).


[KRKNWK-14886]: https://nordicsemi.atlassian.net/browse/KRKNWK-14886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

RobertGalatNordic commented on tests_new/integration/time/src/main.c:
/**
 * split for 2 tests, it checks 2 interfaces, timer and uptime
 * 
 */

RobertGalatNordic commented on tests_new/integration/storage/boards/nrf54l15dk_nrf54l15_cpuapp_ns.overlay:
should't be needed

RobertGalatNordic commented on tests_new/integration/storage/src/main.c:
refactor tests, tests are coupled, need to be run in order, commented test

RobertGalatNordic commented on tests_new/integration/spi/src/main.c:
/**
 * Test spi transfer, reuse external flash because it is already on DK? otherwise this test will require some fixture to run 
 * 
 */

RobertGalatNordic commented on tests_new/integration/mfg/src/main.c:
/**
TODO rewrite the test

read suite:
v7
v8

write suite:

clear memory,
write values
read values
compare


persistant test
write values
dump flash to different address
initialize on new address (copy)
read values
compare

 */

RobertGalatNordic commented on tests_new/integration/crypto/src/main.c:
#if defined(CONFIG_SOC_SERIES_NRF54LX) 
// TODO: this ifdef should be based on feature not SOC used. replace it by feature: SID_PAL_HASH_SHA512 check



 /**
I want to see what configurations were tested. Maybe some kind of matrix that show what function with what algorithms, and what values were tested
what error cases were covered

each function should have its own suite maybe some kind of macro generation of test functions would be usefull to make this information more compact

maybe separating functions, into separate files would be benefitial

maybe some defines, should be moved to Kconfig of the test, or separate configuration header.
Simmilarly there are many schemes like test_data, correct_response. extracting those long data arrays would make tests more redeable.
 */

RobertGalatNordic commented on tests_new/integration/crypto_keys/src/main.c:
// keys_init repeat in every test, move to before_fn ? 

// deinit() repeat ( or should repeat) in all tests move to after_fn ?

RobertGalatNordic commented on tests_new/integration/delay/src/main.c:
// TODO: split to 2 test cases, delay =0, delay > 0, add maybe cases with small delay, and big delay ... test for test for accuracy, print time stamps in fail message (to know why it failed)

RobertGalatNordic replied:
make sure to use correct types, %lld for uint64 ( %d will get lower 32bits, so in values below uint32_max will print 0)

RobertGalatNordic commented on tests_new/integration/interrupts/src/main.c:
add verification of the out buffer before starting swi, and after stop and deinit to check if swi was not triggered too many times, or after stop/deinit

RobertGalatNordic commented on tests_new/integration/mfg/testcase.yaml:
Currently this test fails, and need to be rewritten.
Consider skipping this test, to allow this PR to merge, and plan rewriting it.

RobertGalatNordic reviewed (changes requested):
crypto
mfg
spi
storage

Require rewriting or major refactoring. Please move them to their old location, as they will be addressed in separate PRs.

If some test does not pass, please skip it and place it back in its old location (move to tests_new, only refactored/approved and working tests)

ktaborowski committed:
tests: refactor tests

[KRKNWK-14886]

* Move critical_region test to new location
* Move crypto keys test to new location
* Move crypto test to new location
* Move delay test to new location
* Move interrupts test to new location
* Move spi test to new location
* Move storage test to new location
* Move temperature test to new location
* Move time test to new location

* Add native_sim board support (not all)
* Port to ztest
* Remove sanity tests

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

tests: move mfg storage test

Tests skipped - to be rewritten

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

tests: temperature tests on simulator board

Use mock temperature pal when no temp sensor in dts

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

tests: review

to be squashed

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

tests: fix tests

to be sqashed

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
tests: skip mfg tests

Skip mfg tests for now.
Relace existing tests with simple read write test.

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
tests: remove nrf52 from spi

nrfx spi doesn't pass the tests

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>
